test/libsolidity/syntaxTests/specialFunctions/abidecode/abi_decode_array_of_tuples.sol

```diff
@@ -0,0 +1,13 @@
+contract C {
+    function f() pure public {
+        abi.decode("", ((uint)[2]));
+        abi.decode("", ((uint)[]));
+        abi.decode("", ((uint)[][3]));
+        abi.decode("", ((uint)[4][]));
+        abi.decode("", ((uint)[5][6]));
+        abi.decode("", (((uint))[5][6]));
+        abi.decode("", ((uint, int)[5][6]));
+    }
+}
+// ----
+// TypeError 2614: (301-312): Indexed expression has to be a type, mapping or array (is tuple(type(uint256),type(int256)))
```

Comment:
This test unfortunately has to be split. We get only the error from the first `abi.decode()` and we don't really know if the others really pass or not.

`abi_decode_invalid_argument_tuple_type.sol` has the same problem.

Reply:
Right. I missed those when checking the messages.